packages/file-asset-apis/src/s3/createS3Client.ts

```diff
@@ -1,85 +1,145 @@
 import fs from 'fs-extra';
 import { Agent, AgentOptions as HttpsAgentOptions } from 'https';
-import { S3Client as BaseClient } from '@aws-sdk/client-s3';
+import { S3Client } from '@aws-sdk/client-s3';
 import { NodeHttpHandler, NodeHttpHandlerOptions } from '@smithy/node-http-handler';
-import type { S3ClientConfig as baseConfig } from '@aws-sdk/client-s3';
-import {
-    debugLogger, has, isEmpty, isNumber
-} from '@terascope/utils';
-import type { S3Client } from './client-types';
+import type { S3ClientConfig } from '@aws-sdk/client-s3';
+import { debugLogger } from '@terascope/utils';
+// import type { S3Client } from './client-types';
 
-export interface S3ClientConfig extends baseConfig {
+export interface S3ConnectionConfig {
+    endpoint: string,
+    region: string,
+    forcePathStyle?: boolean,
+    bucketEndpoint?: boolean,
+    accessKeyId?: string
+    secretAccessKey?: string,
+    maxRetries?: number,
     sslEnabled?: boolean,
     certLocation?: string,
-    httpOptions?: HttpsAgentOptions,
-    handlerOptions?: Pick<NodeHttpHandlerOptions, 'requestTimeout'|'connectionTimeout'>,
-    secretAccessKey?: string,
-    accessKeyId?: string
+}
+
+export interface Credentials {
+    accessKeyId: string;
+    secretAccessKey: string;
 }
 
 export async function createS3Client(
-    config: S3ClientConfig,
+    connectionConfig: S3ConnectionConfig,
     logger = debugLogger('s3-client')
 ): Promise<S3Client> {
-    logger.info(`Using S3 endpoint: ${config.endpoint}`);
+    logger.info(`Using S3 endpoint: ${connectionConfig.endpoint}`);
+
+    const clientConfig = await genS3ClientConfig(connectionConfig);
 
     // The aws v3 client logs every request and its metadata, it is too intrusive
     // so should only be used in trace mode otherwise it will log the body request
     // which has heavy performance implications
     if (logger.level() === 10) {
-        config.logger = logger;
+        clientConfig.logger = logger;
+    }
+    logger.debug(`createS3Client finalConfig:\n${JSON.stringify(clientConfig, null, 2)}`);
```

Comment:
I like this debug log, good choice.

Reply:
Imitation is the sincerest form of flattery.